test/event_store/stream_test.exs

```diff
@@ -34,4 +34,70 @@ defmodule Commanded.EventStore.Adapters.Spear.StreamTest do
     assert %TestEvent{name: "foo"} = first
     assert %TestEvent{name: "bar"} = second
   end
+
+  test "parses the system type properly", %{
+    event_store_meta: event_store_meta,
+    event_store_db_uri: event_store_db_uri
+  } do
+    conn = start_link_supervised!({Spear.Connection, [connection_string: event_store_db_uri]})
+
+    insert(event_store_meta, conn, "b0", "teststream-b")
+    insert_link(event_store_meta, conn, {0, "teststream-b"}, "stream_c")
+
+    # wait a bit because the $all projection is not synchronously built
+    :timer.sleep(1000)
```

Comment:
Is there a way to wait to check if the projection has been built? we could wait with a timeout that could be a bit more reliable than a straight sleep.

Reply:
I can wait for the event using a subscription, will do that.